pkgs/development/python-modules/cramjam/default.nix

```diff
@@ -9,33 +9,46 @@
   numpy,
   pytest-xdist,
   pytestCheckHook,
+  pkg-config,
+  isa-l,
+  c-blosc2,
 }:
 
 buildPythonPackage rec {
   pname = "cramjam";
-  version = "2.8.3";
+  version = "2.9.1"; # make sure to update maturinBuildFlags
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "milesgranger";
     repo = "pyrus-cramjam";
     tag = "v${version}";
-    hash = "sha256-1KD5/oZjfdXav1ZByQoyyiDSzbmY4VJsSJg/FtUFdDE=";
+    hash = "sha256-s+dKoZftI+aXcie1quAo/Xmlt1BQrzoCjwesDiaNABY=";
   };
 
   cargoDeps = rustPlatform.fetchCargoVendor {
     inherit src;
-    hash = "sha256-Gpj/LUCx/ivYlmxNJnEZr8caEfV4OaQwEPNjz7vobsw=";
+    name = "${pname}-${version}";
+    hash = "sha256-xrqlVgAUTsqz7AOQdPCSxjRd/Z3ZVpr0qphw6qN6Wys=";
   };
 
-  buildAndTestSubdir = "cramjam-python";
-
   nativeBuildInputs = with rustPlatform; [
     cargoSetupHook
     maturinBuildHook
+    pkg-config
   ];
 
-  buildInputs = lib.optional stdenv.hostPlatform.isDarwin libiconv;
+  buildInputs = [
+    isa-l
+    c-blosc2
+  ] ++ lib.optional stdenv.hostPlatform.isDarwin libiconv;
+
+  maturinBuildFlags = [
+    "--no-default-features"
+    "--features"
+    # default features from Cargo.toml, modified to specify -shared
+    "extension-module,snappy,lz4,bzip2,brotli,xz,zstd,gzip,zlib,deflate,blosc2-shared,use-system-isal-shared,igzip-shared,ideflate-shared,izlib-shared,use-system-blosc2-shared"
```

Comment:
Those marked with -shared here become experimental/optional on the master. Maybe, we don't need to ship those. Why xz and others don't need to be -shared?

Reply:
Oh, you are right that they are disabled by default now. I suppose we can do that for this version already.

About xz and the others, I'm not 100% sure. Maybe they are always shared. I'll check.